### PR TITLE
added basic networkconfig resource

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/etcdv2/fake/fake.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package fake
+
+import (
+	"context"
+)
+
+type Fake struct {
+}
+
+func New() *Fake {
```
Comment (xh3b4sd) [inline service/etcdv2/fake/fake.go]: We use this just as stub for testing.

File: service/etcdv2/spec.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package etcdv2
+
+import (
+	"context"
+)
+
+type Store interface {
```
Comment (xh3b4sd) [inline service/etcdv2/spec.go]: For testing we need to configure an interface. We can then replace it with the fake implementation.

File: service/key/key.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package key
```
Comment (xh3b4sd) [inline service/key/key.go]: Here are the key helpers the migrated resources use.

File: service/resource/legacy/resource.go [hunk 4]
```diff
@@ -112,60 +102,6 @@ func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desire
 		spec = o.Spec
 	}
 
-	// Create flannel etcd config.
```
Comment (xh3b4sd) [inline service/resource/legacy/resource.go]: Here we can now remove the etcd magic because it is done in the separate resource.

**Comment by xh3b4sd** [comment]
Successfully tested on lycan.

File: service/key/key.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+func EtcdNetworkPath(customObject flanneltpr.CustomObject) string {
+	return "coreos.com/network/" + NetworkBridgeName(customObject)
+}
+
+func NetworkBridgeName(customObject flanneltpr.CustomObject) string {
+	return "br-" + ClusterID(customObject)
+}
+
+func ToCustomObject(v interface{}) (flanneltpr.CustomObject, error) {
```
Comment (rossf7) [inline service/key/key.go]: <3

File: service/resource/networkconfig/update.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	// We do not compute the create state here because the framework already
+	// handles it separately.
+	var networkConfigToCreate NetworkConfig
+
+	// We do not compute the delete state here because there are no resources we
+	// can safely remove.
```
Comment (rossf7) [inline service/resource/networkconfig/update.go]: These comments are very clear and useful <3

**Comment by xh3b4sd** [comment]
I saw that in Joe's PRs in the prometheus config controller repo. I thought this makes sense and keeps the scope of the files down. It is "unusual" but we can make it a convention. I like it. It makes sense to me.

**Pull request merged**